astropy/cosmology/flrw/base.py

```diff
@@ -84,16 +84,20 @@ def scale_factor0(self):
         """
         return u.Quantity(self.scale_factor(0), unit=u.one)
 
+    @deprecated_keywords("z", since="7.0")
     def scale_factor(self, z):
```

Comment:
Haven't looked through the C-code yet, but does `@deprecated_keywords` map to the positions? As in, can this be made positional-only now? And then we remove the decorator later, to complete the deprecation?

Reply:
It doesn't. Maybe it's possible but I don't know the C API well enough to see how it could be implemented.

Reply:
Actually I *do* think it's feasible, but it's a trade-of for where we put complexity: it would certainly make the C extension more complex and I'm already pushing my own limit there so I'm not confident that I could build this feature without hurting robustness and readability. How about I give it a try as a follow up task ?

Reply:
I think that's totally reasonable!  And thanks again for taking the lead on this!
Having the parameters be positional-only at the undecorated Python function level is an improvement, but to the user there is no difference, so the PR as written is great 👍.

Reply:
dully noted !

Reply:
In hindsight and given the complaints this PR got for introducing a hard-to-maintain C extension, I'm not sure it's worth pushing it any further without a clear need to.